src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopExe.cs

```diff
@@ -33,13 +33,31 @@ public void It_builds_a_simple_desktop_app()
             });
         }
 
+        [Fact]
```

Comment:
Do you have any way to future proof this test? I was thinking essentially something long the lines of 

```csharp
[Theory]
[MemberData(DataStorage.Tfms)]
public void Gandalf_excess_references_to_the_compiler(string tfm
{
        var testProject = new TestProject()
        {
            TargetFrameworks = tfm,
            IsExe = true,
            ProjectSdk = "Microsoft.NET.Sdk"
        };
  ...
}
```

Then in a more central place

```csharp

    public static IEnumerable<object[]> Tfms =>
        new List<object[]>
        {
            new object[] { "net7.0" },
            new object[] { "net8.0" },
        };
```

This way when you release the `net9.0` TFM you update a single place and a number of tests get verified on the new TFM.

Reply:
Yeah, we built this for the linker/AOT tests. If you just want the current TFM you can do [ToolsetInfo.CurrentTargetFramework](https://github.com/dotnet/sdk/blob/12a7cfd34eb17d8faf8f70e53b20369390f367a3/test/Microsoft.NET.TestFramework/ToolsetInfo.cs#L11)

If you want all supported TFMs you can use [PublishTestTools.SupportedTfms](https://github.com/dotnet/sdk/blob/12a7cfd34eb17d8faf8f70e53b20369390f367a3/test/Microsoft.NET.Publish.Tests/PublishTestUtils.cs#L10). Don't ask me why they're in separate places.

Reply:
> Don't ask me why they're in separate places.

Good feedback, will log a refactoring issue for domestic cat to pickup in spare cycles.